gabxgonz/cat-game-collab
# Adding flexibility to interactions with UsePaw, UseMouth, and UseBody

# How to test: 
See `irritation-test` scene to test it out!

For a new object, add the interaction UI Controller script to an interactable object to set the interaction option. It will also need the Interactable Controller as well.

You can set multiple interactions and set the text to anything you'd like! (such as "Scoot butt across clean carpet")

![image](https://user-images.githubusercontent.com/21282705/133009815-044df69e-e35d-4a1e-b32c-053ef1f70814.png)

![image](https://user-images.githubusercontent.com/21282705/133009762-6eb9b326-152a-4ce9-96ad-b3606d098341.png)

# Notes:

- Actions aren't yet limited to the ones set on the interactable object (so for example if only "use paw" is available, pressing T will still have an effect.)
- There are also some placeholders to connect up animations later on.
- The UI is visually a little rough still, maybe we can come up with a cuter design for the NPC irritation bars & tool tips
